### PR TITLE
Set custom SentryHttpClientError when HTTP error is captured without an exception

### LINKED ISSUE #579: null `type` and `value` exception for the `SentryHttpClient` errors
if you force an error by changing the https://github.com/getsentry/sentry-mobile/blob/main/lib/api/sentry_api.dart#L44-L45 URL from `organizations` to `organization`, you get 2 errors in Sentry.

1st) ApiError, triggered by the Mobile App itself
2nd) Automatically triggered by SentryHttpClient

the 2nd one has no `type` and `value`, See:

Screenshot 2021-09-03 at 11 20 43

I'd like to know if the bug is Sentry Mobile App throwing a type that does not have type and value at runtime or SentryHttpClient from the Sentry SDK is failing to detect it automatically.

### PR DESCRIPTION
**Pull request by marandaneto**
## :bulb: Motivation and Context
Fix https://github.com/getsentry/sentry-dart/issues/579

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
Screenshot 2021-09-03 at 14 44 47

File: dart/lib/src/http_client/sentry_http_client_error.dart [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class SentryHttpClientError extends Error {
```
[the remark this answers is not in the document]
Comment (marandaneto) [reply dart/lib/src/http_client/sentry_http_client_error.dart]: good one, also changed `ThrowableMechanism`

File: dart/test/mocks/mock_transport.dart [hunk 2]
```diff
@@ -1,17 +1,63 @@
...
+    envelopeMap['request']['headers'] = newHeadersMap;
+
+    final otherMap = requestJson?['other'] as Map<String, dynamic>?;
+    final newOtherMap = <String, String>{};
+    if (otherMap != null) {
+      for (final entry in otherMap.entries) {
+        newOtherMap[entry.key] = entry.value as String;
+      }
+    }
+    envelopeMap['request']['other'] = newOtherMap;
```
Comment (marandaneto) [inline dart/test/mocks/mock_transport.dart]: `jsonDecode` causes this

File: dart/lib/src/throwable_mechanism.dart [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import 'protocol/mechanism.dart';
 
-/// An Error decorator that holds a Mechanism related to the decorated Error
-class ThrowableMechanism extends Error {
+/// An Error decorator that holds a Mechanism related to the decorated Exception
```
Comment (ueman) [inline dart/lib/src/throwable_mechanism.dart]:
```suggestion
/// An Exception decorator that holds a Mechanism related to the decorated Exception
```

Comment (ueman) [reply dart/lib/src/throwable_mechanism.dart]: or just
```suggestion
/// A decorator that holds a Mechanism related to the decorated Exception
```

**Review by ueman — APPROVED** [review]
LGTM, with a small nitpick

**Pull request merged**